src/main/scala/algolia/definitions/WaitForTaskDefinition.scala

```diff
@@ -64,6 +60,19 @@ trait WaitForTaskDsl {
   implicit object WaitForTaskDefinitionExecutable
       extends Executable[WaitForTaskDefinition, TaskStatus] {
 
+    // Run every 50 ms, use a wheel with 512 buckets
+    private lazy val timer = {
+      val threadFactory = new ThreadFactory {
```

Comment:
Actually the whole point being to reuse this `HashedWheelTimer` for all submitted queries, there is no way to use the query itself to create it.

Given that it is used for network calls, I think 50 or 100 ms is a good number, especially considering the default base delay is 100ms and then 200, 400 etc.

What do you think?